packages/connect/src/api/ethereum/api/ethereumSignTypedData.ts

```diff
@@ -226,8 +226,14 @@ export default class EthereumSignTypedData extends AbstractMethod<'ethereumSignT
                     const memberTypeDefinition = types[memberTypeName][index];
                     memberTypeName = memberTypeDefinition.type;
                     memberData = memberData[memberTypeDefinition.name as keyof typeof memberData];
-                } else {
-                    // TODO: what to do when the value is missing (for example in recursive types)?
+                }
+                if (memberData === null || memberData === undefined) {
+                    // Cancel the request so the device isn't left hanging
+                    cmd.typedCall('Cancel', 'Success');
```

Comment:
hmm I just wonder - shoudln't this be handled somewhere in a general manner? I would even say there are some attempts to have it cover unfortunately they are a mess. https://github.com/trezor/trezor-suite/blob/develop/packages/connect/src/device/DeviceCommands.ts#L101

and this is something @marekrjpolak is now looking into

Reply:
ok, we probably don't want to introduce ethereum specific to where general logic is handled so this is probably the fastest way to do it now.